### ISSUE TITLE
Add NodeDataModel "slots" for connections added/removed

### ISSUE THREAD

**Issue by Quincunx271** [paceholder/nodeeditor#174]
Currently, if I have a `MyDataModel : NodeDataModel`, I can be informed when the input data changes with `setInData(...)`, and I can be informed when someone requests my output data with `outData(...)`. However, I've encountered a need to know just when new connections are added. I'm proposing adding:

```c++
virtual
void
inputConnectionAdded(Connection const&);

virtual
void
inputConnectionRemoved(Connection const&);

virtual
void
outputConnectionAdded(Connection const&);

virtual
void
outputConnectionRemoved(Connection const&);
```

The exact signatures / names aren't that important, but it's important that the passed information includes:

* The port it's being connected to.
* The other node and its port that we are being connected to.

**Comment by paceholder** [comment]
As I understand, you want to add the functions to the `NodeDataModel` and trigger them from the `Node`? This makes sense. Could you please go ahead and PR it?

Thanks

Dmitry

**Comment by Quincunx271** [comment]
I found a workaround: `connect(this, &FlowScene::signal, this, [this](Connection const& c) { slot(c); });` instead of `connect(this, &FlowScene::signal, this, &FlowScene::slot);`

**Comment by paceholder** [comment]
Merged as #211
